### ISSUE TITLE
Not working with Nova Flexible Content outside of resource

### ISSUE THREAD

**Issue by eboye** [phoenix-lib/nova-nested-tree-attach-many#22]
I'm using [Nova Flexible Content](https://github.com/whitecube/nova-flexible-content) together with [Nova Settings](https://github.com/optimistdigital/nova-settings) library, and when I tried to add NestedTreeAttachManyField field it shows the field, but throws this error:

https://flareapp.io/share/xPQzjQr7#F1

I'm not exactly sure why this is happening. I'm trying to debug, but if someone has any ideas it would be helpful.

Thanks

**Comment by phoenix-lib** [comment]
@eboye. Hi as I see - Nova Flexible Content modify param 'attribute' and concatenate with hash. This is done to eliminate duplicate fields. Need time for tests and think.

**Comment by eboye** [comment]
Yup, it actually makes json object in database and uses all sorts of hashing and changing of field names to avoid collisions.

Thanks for looking into it. My backup plan is to use searchable select for this if this one doesn't work. But it's not as good as yours.

**Comment by eboye** [comment]
I believe it's this line that is causing this Exception

https://github.com/phoenix-lib/nova-nested-tree-attach-many/blob/4a7a1ffa36b20a4a49b2bd32a859bb4dbe0d39fe/src/NestedTreeAttachManyField.php#L37

The attribute is assumed to be the actual column in database where in Flexible it's not.

So I think it's better to access it via $resource and have it in plural as it's the "standard way of defining relations".

Or better yet, make a fallback where if there is no relation with attribute you fallback to the resource based.

**Comment by phoenix-lib** [comment]
@eboye - Hi, today released version 1.8.0 and now package can be used with Nova Flexible Content. Example of use in readme.md

**Comment by eboye** [comment]
I've tested it and it works, thanks!

But ... :)

On details view, when using single select it shows all possible options and not the chosen one, haven't tested the index view.

Also for some reason it doesn't show up at all when used in [nova-settings](https://github.com/optimistdigital/nova-settings). But that may be another issue :)

details:

![image](https://user-images.githubusercontent.com/624357/132145207-e79d6731-ae0c-46d1-8299-7ea60b00b9ef.png)

forms:

![image](https://user-images.githubusercontent.com/624357/132145218-faf5276d-72bd-4775-9b0c-89fcc7c36834.png)


```
            NestedTreeAttachManyField::make( 'Top Level Category', 'top_level_category', \App\Nova\Collection::class )
                                     ->useSingleSelect()
                                     ->searchable( true )
                                     ->withActiveKey( 'is_top_collection' )
                                     ->useAsField()
                                     ->withAlwaysOpen( true )
                                     ->required(),
```
